flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/reader/KafkaPartitionSplitReader.java

```diff
@@ -75,15 +81,25 @@
     private final SimpleCollector<T> collector;
     private final String groupId;
     private final int subtaskId;
+
+    // For Kafka specific metrics
     private final KafkaSourceReaderMetrics kafkaSourceReaderMetrics;
 
+    // For FLIP-33 standard metrics
+    private final SourceReaderMetricGroup sourceReaderMetricGroup;
+    @Nullable private final Supplier<Long> bytesConsumedTotalSupplier;
```

Comment:
I don't immediately see the value of using the `Supplier` indirection over using `@Nullable Metric bytesConsumedMetric`. Having a nullable supplier is kind of an anti-pattern as we are mixing functional-style with imperative constructs (null).